source/libraries/HelperFunctions.ts

```diff
@@ -71,3 +74,11 @@ export async function initializeTestRpcClientOptions(ethHttpProviderPort: number
 
     return { gasLimit: GAS_BLOCK_AMOUNT, accounts: accountOptions };
 }
+
+export async function waitForTransactionToBeSealed(ethjsQuery: EthjsQuery, transactionHash: string): Promise<void> {
+    let transaction = await ethjsQuery.getTransactionByHash(transactionHash);
+    while (transaction.blockNumber == null) {
+        await sleep.sleep(5);
```

Comment:
5 seconds is _way_ too long.  This should probably be something on the order of 100 milliseconds or less, maybe even 10 milliseconds.

Reply:
If the block interval is 10-15 seconds, a 10-100ms retry will result in a lot of fast, failed retries.  I guess for something like instantseal it should be faster though.

Reply:
We shouldn't assume a particular block time because when integration testing we can set the block time to whatever we want.  The only reason for the sleep at all is just so we don't melt the CPU while spinning.  I'm guessing 10ms is enough of a sleep to keep CPUs cool, hence the recommendation.

Reply:
This function will be called when we use the uploader on a live chain, not just when we're using it for integration testing / against instantseal Parity or testrpc.  Each iteration of this loop sends off an RPC request to the chain.  Each RPC request is small, but if you send one out every 10ms and the mainnet or Ropsten block interval is 15s, you'll send out on average 15000 RPC requests per transaction.  That represents a significant amount of network I/O and should be avoided if possible IMO.

Reply:
Are we planning on using this script against a remote chain?  15,000 requests against a local node on a modern computer shouldn't be too bad.  I'm amenable to 100ms (10 requests/second) if people are finding their computers to be catching on fire though.

Reply:
I think it's sensible to make it usable with a remote chain.  I guess I don't see the importance of a short sleep time here.  If you're using InstantSeal or testrpc the sleep shouldn't happen at all, since sealing happens immediately.  It could be annoying if you're using a private chain without InstantSeal, but with a shorter block interval.  What we could do is make it read from an environment variable `process.env.ETHEREUM_BLOCK_INTERVAL_MILLISECONDS` and if that variable isn't set default to reasonable values for the main chain (5 seconds is reasonable IMO).

Reply:
Geth doesn't have instaseal, and even with Parity its a race as to whether it seals before or after you make the following `eth_call` since Parity is multithreaded.

I'm 👍 on an environment variable, though I would either name the variable `ETHEREUM_BLOCK_TIME_IN_MILLISECONDS` and have it be 15 seconds for main network or name the variable `ETHEREUM_POLLING_INTERVAL_IN_MILLISECONDS` and have it set to the desired polling interval.  I vote is for the latter, but I'm fine with the former as well (and then internally just have polling time be `block_time / 10` or something.